compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -4531,12 +4538,47 @@ abstract class GenJSCode extends plugins.PluginComponent
     sym.isAnonymousClass && AllJSFunctionClasses.exists(sym isSubClass _)
 
   private def isRawJSCtorDefaultParam(sym: Symbol) = {
+    isCtorDefaultParam(sym) &&
+    isRawJSType(patchedLinkedClassOfClass(sym.owner).tpe) &&
+    nme.defaultGetterToMethod(sym.name) == nme.CONSTRUCTOR
+  }
+
+  private def isJSNativeCtorDefaultParam(sym: Symbol) = {
+    lazy val companionClass = patchedLinkedClassOfClass(sym.owner)
+    isCtorDefaultParam(sym) &&
+    isRawJSType(companionClass.tpe) &&
+    !isScalaJSDefinedJSClass(companionClass)
```

Comment:
As this pattern already appears 3 times I will create `private def isJSNativeClass(sym: Symbol): Boolean` which is even more verbose and intuitive.